x-pack/plugins/advanced_ui_actions/public/components/action_wizard/action_wizard.tsx

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState } from 'react';
+import {
+  EuiButtonEmpty,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiIcon,
+  EuiSpacer,
+  EuiText,
+  EuiKeyPadMenuItem,
+} from '@elastic/eui';
+import { txtChangeButton } from './i18n';
+import './action_wizard.scss';
+
+// TODO: this interface is temporary for just moving forward with the component
+// and it will be imported from the ../ui_actions when implemented properly
+// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
+export type ActionFactoryBaseConfig = {};
+export interface ActionFactory<Config extends ActionFactoryBaseConfig> {
+  type: string;
+  displayName: string;
+  iconType?: string;
+  wizard: React.FC<ActionFactoryWizardProps<Config>>;
+}
+
+export interface ActionFactoryWizardProps<Config extends ActionFactoryBaseConfig> {
+  /**
+   * Current (latest) config of the item. (state)
+   */
+  config: Config | null;
+
+  /**
+   * Callback called when user updates the config in UI.
+   * ActionFactory's wizard should do validations to the user's input
+   * In case input is complete and valid - config: Config object should be emitted
+   * In case input has changed to the invalid state: null should be emitted
+   */
+  onConfig: (config: Config | null) => void;
+}
+
+export interface ActionWizardProps {
+  /**
+   * List of available action factories
+   */
+  actionFactories: Array<ActionFactory<ActionFactoryBaseConfig>>;
+
+  /**
+   * Notifies when wizard's state changes because of user's interaction
+   *
+   * @param actionFactory - current selected action factory. null if none is selected
+   * @param config - current config for current action factory. null if no action factory or if wizard's inputs are invalid or incomplete
+   */
+  onChange: (
+    actionFactory: ActionFactory<ActionFactoryBaseConfig> | null,
+    config: ActionFactoryBaseConfig | null
```

Comment:
idea is same as here - https://github.com/elastic/kibana/pull/59032/files#r387845734

Reply:
actionFactory at least I think could be `?` instead of `| null`

Gave another option for using undefined (or requiring config) above instead of null, maybe it'll work.

Reply:
for passing data down went with `?` 👍 
for reseting data with callbacks left `null`. it seems semantically better to me